### PR TITLE
DB Access: OnChange cache

### PR DESCRIPTION
**Pull request by sachinholla**
1\) Introduced the OnChange cache in the DB object. Cache is maintained as map[string]db.Value, where map key is the redis key.

OnChange cache is not enabled by default. DB instance should be inited with IsEnableOnChange=true option and caching should be enabled for specific tables by calling DB.RegisterTableForOnChangeCaching(). DB.OnChangeCacheUpdate() and DB.OnChangeCacheDelete() functions should be used to refresh the cache contents.

2\) Enhanced DB.GetEntry() to return value from the OnChnage cahce if the key exists in the cache. Otherwise the value will be read from redis and cache will be updated.

Part of the subscription enhancements described in the HLD https://github.com/sonic-net/SONiC/pull/1287

### CODE REVIEW

File: translib/db/db_onchangecache.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+type dbOnChangeReg struct {
+	CacheTables map[string]bool // Only cache these tables.
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//  Exported Functions                                                        //
+////////////////////////////////////////////////////////////////////////////////
+
+func (d *DB) RegisterTableForOnChangeCaching(ts *TableSpec) error {
+	if glog.V(1) {
```
Comment (tomek-US) [inline translib/db/db_onchangecache.go:39]: Is this message _really_ that important to have level set to 1? As I read the code it is a debugging message, so, level 3 would look good enough.

File: translib/db/db.go [hunk 2]
```diff
@@ -156,7 +156,8 @@ type Options struct {
 	InitIndicator      string
 	TableNameSeparator string //Overriden by the DB config file's separator.
 	KeySeparator       string //Overriden by the DB config file's separator.
-	IsWriteDisabled    bool //Indicated if write is allowed
+	IsWriteDisabled    bool   //Indicated if write is allowed
+	IsEnableOnChange   bool   // whether OnChange cache enabled
```
[the remark this answers is not in the document]
Comment (sachinholla) [reply translib/db/db.go:160]: We want have the OnChange cache disabled by default.. It should be enabled only for the DB objects crated by `trannslib.Subscribe()` function. Hence cannot name this option as `OnChangeDisabled`.

@a-barboza, shall I rename as `IsOnChangeEnabled` ?

[the remark this answers is not in the document]
Comment (a-barboza) [reply translib/db/db.go:160]: @sachinholla No Objection

**Review by tomek-US — APPROVED** [review]
LGTM

**Pull request merged**